src/components/Swipeable/index.tsx

```diff
@@ -0,0 +1,41 @@
+import {
+  createContext,
+  PropsWithChildren,
+  useCallback,
+  useContext,
+  useRef,
+} from "react";
+import SwipeableLib from "react-native-gesture-handler/Swipeable";
+import Like from "./Like";
+
+type SwipeableContextType = {
+  onOpenNextSwipeable: (next: React.RefObject<SwipeableLib | null>) => void;
+};
+
+const SwipeableContext = createContext<SwipeableContextType>({
+  onOpenNextSwipeable: () => {},
+});
+
+export const Provider = ({ children }: PropsWithChildren) => {
```

Comment:
I think you can achieve this logic without context - by having `ref` defined in PokemonList and passed it to each single `SwipeableLike`.